text/0000-remove-shell-processes.md

```diff
@@ -0,0 +1,468 @@
+
+### Evaluating Args
+
+For shell processes provided by older buildpacks, we must continue to Bash evaluate buildpack-provided args to avoid breaking older buildpacks
+
+For example, if a buildpack using API `0.5` creates the following entry in `launch.toml`
+```
+[[process]]
+type = "hi"
+command = "echo"
+args = ["hello", "${WORLD:-world}"]
+```
+
+It should be converted to the following in `metadata.toml`
+
+```
+[[process]]
+type = "hi"
+command = ["echo", "hello", "${WORLD:-world}"]
+args = []
+api = "0.5"
+direct = false
+```
+
+and the launcher should Bash evaluate each entry in `command` to avoid breaking changes.
+
+However, any additional user-provided args should NOT be Bash evaluated to reduce the amount of complexity exposed to end users.
+
+### profiles
+Profiles contributed by older buildpacks will still be evaluated when executing shell process types. But user provided profiles will not be evaluated when using the new platform API, even when a shell process is executed. Again, this is done to prevent differences between the buildpack APIs from leaking into the user interface. Users will only need think about differences between platform APIs.
+
+## Migration Path
+
+### Buildpack Authors
+When migrating to the new API, buildpack authors should take the following steps:
+1. Does the buildpack contribute any profile.d helpers? If so, replace these with equivalent exec.d helpers.
+1. Does the buildpack contribute a `direct=true` process?  If so, remove `direct=true` from the process definition, this is now the default.
+1. Does the buildpack contribute a `direct=false` process? If so, there are two options:
+   1. Explicitly add `bash` or `cmd` to the process definition, this is the safest path forward.
+   2. Remove an unnecessary dependency on  `bash` or `cmd` by:
+        * Changing any environment variable references to use the `$(<env)` syntax.
+        * Ensure that your process functions properly as PID1.
+
+## Platform Maintainers
+
+Before upgrading to the new API platform, platforms that wish to support the `<app>/.profile` or `<app>/.profile.bat` should ensure that all builds contain a buildpack that provides support for this feature, whenever the stack provides the requisite shell.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+In exchange for a reduction in complexity and cognitive overhead buildpack-authors and end users lose certain conveniences like the more intuitive profile interface. This drawback could be remediated by tooling for buildpack authors (to make creation of `exec.d` helpers easy) or by buildpacks (to support `.profile`). However, in the `.profile` case, consistency across ecosystems will be a matter of convention (like Procfile) rather than a guarantee.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## `$((<env>))` syntax
+
+We could use our own explicit syntax like `$((<env>))` or environment variable replacements instead of `$(<env>)`.
+pros:
```

Comment:
The pros/cons look funny in the PR markdown preview. I think you may need an additional line break between lists